sdk/core/src/bytes_response.rs

```diff
@@ -11,81 +9,90 @@ pub(crate) struct BytesResponse {
 
-        Ok((response, bytes_response))
-    }
+            let response = Response::new(
+                status_code,
+                header_map.clone(),
+                Box::pin(BytesStream::new(response_bytes.clone())),
+            );
+            let bytes_response = BytesResponse::new(status_code, header_map, response_bytes);
 
-    pub(crate) fn deconstruct(self) -> (StatusCode, HeaderMap, Bytes) {
-        (self.status, self.headers, self.body)
+            Ok((response, bytes_response))
+        }
     }
-}
 
-impl<'de> Deserialize<'de> for BytesResponse {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::de::Deserializer<'de>,
-    {
-        use serde::de::Error;
-        let r = SerializedBytesResponse::deserialize(deserializer)?;
-        let mut headers = HeaderMap::new();
-        for (n, v) in r.headers.iter() {
-            let name = header::HeaderName::from_lowercase(n.as_bytes()).map_err(Error::custom)?;
-            let value = header::HeaderValue::from_str(&v).map_err(Error::custom)?;
-            headers.insert(name, value);
-        }
-        let body = Bytes::from(base64::decode(r.body).map_err(Error::custom)?);
-        let status = StatusCode::from_u16(r.status).map_err(Error::custom)?;
+    impl<'de> Deserialize<'de> for BytesResponse {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::de::Deserializer<'de>,
+        {
+            use serde::de::Error;
+            let r = SerializedBytesResponse::deserialize(deserializer)?;
+            let mut headers = HeaderMap::new();
+            for (n, v) in r.headers.iter() {
+                let name =
+                    header::HeaderName::from_lowercase(n.as_bytes()).map_err(Error::custom)?;
+                let value = header::HeaderValue::from_str(&v).map_err(Error::custom)?;
+                headers.insert(name, value);
+            }
+            let body = Bytes::from(base64::decode(r.body).map_err(Error::custom)?);
+            let status = StatusCode::from_u16(r.status).map_err(Error::custom)?;
 
-        Ok(Self::new(status, headers, body))
+            Ok(Self::new(status, headers, body))
+        }
     }
-}
 
-impl Serialize for BytesResponse {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::ser::Serializer,
-    {
-        let mut headers = HashMap::new();
-        for (h, v) in self.headers.iter() {
-            headers.insert(h.as_str().into(), v.to_str().unwrap().into());
+    impl Serialize for BytesResponse {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::ser::Serializer,
+        {
+            let mut headers = HashMap::new();
+            for (h, v) in self.headers.iter() {
+                headers.insert(h.as_str().into(), v.to_str().unwrap().into());
```

Comment:
Are we sure this `unwrap` will never panic?

Reply:
No, we could store the header value as bytes if we want to. This isn't a change made in this PR so I won't block merging on it though.

cc @MindFlavor